enhanced/enhanced.go

```diff
@@ -190,7 +192,7 @@ func (e *Exporter) collectValue(ch chan<- prometheus.Metric, instance config.Ins
 			}
 			metrics, ok := u.(map[string]interface{})
 			if !ok {
-				return fmt.Errorf("%s: wrong value type for metrics: %T", key, metrics)
+				log.Error(fmt.Errorf("%s: wrong value type for metrics: %T", key, metrics))
```

Comment:
it's not return any error, to function or stdout, it's just silently dies

Reply:
@arvenil sometimes parser can meet storage string in some metric.
we shouldn't stop all metrics processing in case if one metric in the middle has the wrong format

Reply:
@delgod And it's fine, as long as you verified that it won't lead to panics or crashes because some of those returns might be there for a reason to NOT continue further processing, e.g.
```patch

		if len(extraLabels) != len(extraLabelsValues) {
-			return fmt.Errorf("%s: len(labels) != len(labelsValues) for metric %s: len(%T) != len(%T)", key, metricName, extraLabels, extraLabelsValues)
+			log.Error(fmt.Errorf("%s: len(labels) != len(labelsValues) for metric %s: len(%T) != len(%T)", key, metricName, extraLabels, extraLabelsValues))
		}
		e.sendMetric(ch, instance, namespace, subsystem, name, metricValue, extraLabelsValues...)
```
When this condition is true the e.sendMetric command should never be called as prometheus with wrong number of labels will crash the client.

Also this example is also wrong because you will send metric with value `0` but if we can't parse this metric we shouldn't send it at all - why assume it's `0` when we can't read it?

Reply:
the issue, when maps arrives, sometimes, it got string first and stop process results, which leads to gap in metrics